server/src/main/java/org/cloudfoundry/identity/uaa/oauth/KeyInfo.java

```diff
@@ -285,7 +306,16 @@ public Map<String, Object> getJwkMap() {
         result.put("use", JsonWebKey.KeyUse.sig.name());
         result.put("kid", this.keyId);
         result.put("kty", RSA.name());
-
+        // X509 releated values from JWK spec
+        if (this.verifierCertificate != null) {
+            X509Certificate x509Certificate = X509CertUtils.parse(verifierCertificate);
+            if (x509Certificate != null) {
+                byte[] encoded = JwtHelper.getX509CertEncoded(x509Certificate);
+                result.put("x5c", Collections.singletonList(Base64.encode(encoded).toString()));
+                result.put("x5t", JwtHelper.getX509CertThumbprint(encoded, "SHA-1"));
+                result.put("x5t#S256", JwtHelper.getX509CertThumbprint(encoded, "SHA-256"));
```

Comment:
I don't understand what `x5t#S256` is.

Reply:
hash with sha-256 instead sha-1 , see rfc 7517